### PR TITLE
vfep-1571: Form-1990 Allow only one benefit selection per submission

### PR DESCRIPTION
**Pull request by fatmakhan0395**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- _(Summarize the changes that have been made to the platform)_
- Updated checkboxes to radio buttons to only one benefit selection per submission and added a new bullet point in 1990 form. The review page is also updated with radio inputs.
- Updated unit tests
- Updated error msg.

- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_ Vfep, YES
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
As a...

VFEP Business User
 
I want...
to select the benefit I want on the 22-1990

So that...

I can be evaluated for the benefit 
 
Acceptance Criteria:
 
The user is presented with benefit options and as opposed to being able to select any combination of these options, they can only select one.  This may involve changing the checkboxes to radio buttons.  The selection shall still be required:

image

The gray box above the benefit selection has a new bullet added after "You may be eligible for more than one program".  The new bullet states "If you wish to apply for more than one benefit, submit another 22-1990 application."

image

Technical Details:
Feature Flag needed: Yes
UAT needed: Yes
 
Definition of Done: 
(  )  Passed all internal testing
(  )  Demo/UAT completed with EDU
(  )  All work documented in Jira ticket
(  )  Added Feature flag/toggles

department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

https://jira.devops.va.gov/browse/VFEP-1571

## Testing done

- _Describe what the old behavior was prior to the change_ Previously, a user could select multiple options for benefits eligibility.
- _Describe the steps required to verify your changes are working as expected_ Navigate to step 2 of 5 Benefits eligibility page on 1990 form. Select one of the radio options and continue.
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

Before:
image

After:
image

image
image

Mobile
image

image

image

## What areas of the site does it impact? 1990 form, benefits eligibility page.

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  chapter1606,
+) => {
+  const uiSchema = isProductionOfTestProdEnv()
+    ? {
+        'ui:description': benefitsEligibilityBox,
+        'view:selectedBenefits': {
+          'ui:title': 'Select the benefit that is the best match for you.',
+          'ui:validations': [validateBooleanGroup],
+          'ui:errorMessages': {
+            atLeastOne: 'Please select at least one benefit',
```
Comment (jsimonVA) [inline src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js:18]: Now that we are making this a radio button that will only allow 1 selection, the error message should be updated. My recommendation is: required: "You must select a benefit"

File: src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            'ui:title': benefitsLabels.chapter1606,
+          },
+        },
+      }
+    : {
+        'ui:description': benefitsEligibilityBox,
+        'view:selectedBenefits': {
+          'ui:title': 'Select the benefit that is the best match for you.',
+          'ui:errorMessages': {
+            atLeastOne: 'Please select at least one benefit',
```
Comment (jsimonVA) [inline src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js:42]: Now that we are making this a radio button that will only allow 1 selection, the error message should be updated. My recommendation is: required: "You must select a benefit"

File: src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          },
+          chapter33: {
+            'ui:widget': 'radio',
+            'ui:required': formData => !formData.chapter33,
+            'ui:options': {
+              labels: benefitsLabels,
+              hideLabelText: true,
+            },
+            'ui:errorMessages': {
+              required: 'Please select at least one benefit',
```
Comment (jsimonVA) [inline src/applications/edu-benefits/1990/pages/benefitsEligibilityUpdate.js:55]: Now that we are making this a radio button that will only allow 1 selection, the error message should be updated. My recommendation is: required: "You must select a benefit"

**Review by GovNapoleon — APPROVED** [review]

**Review by jsimonVA — APPROVED** [review]
Looks great.

**Pull request merged**